cmd/pfcpsim-client/main.go

```diff
@@ -3,6 +3,362 @@
+// Returns error if fail occurs at any stage.
+func getLocalAddress() (net.IP, error) {
+	cmd := "route -n get default | grep 'interface:' | grep -o '[^ ]*$'"
+	cmdOutput, err := exec.Command("bash", "-c", cmd).Output()
+	if err != nil {
+		return nil, err
+	}
+
+	interfaceName := strings.TrimSuffix(string(cmdOutput[:]), "\n")
+
+	itf, _ := net.InterfaceByName(interfaceName)
+	item, _ := itf.Addrs()
+	var ip net.IP
+	for _, addr := range item {
+		switch v := addr.(type) {
+		case *net.IPNet:
+			if v.IP.To4() != nil { //Verify if IP is IPV4
+				ip = v.IP
+			}
+		}
+	}
+
+	if ip != nil {
+		return ip, nil
+	}
+
+	return nil, fmt.Errorf("could not find interface: %v", interfaceName)
+}
+
+// parseArgs perform flag parsing and validation saving necessary data to global variables.
+func parseArgs() {
+	inputF := getopt.StringLong("input-file", 'f', "", "File to poll for input commands. Default is stdin")
+	outputFile := getopt.StringLong("output-file", 'o', "", "File in which copy from Stdout. Default uses only Stdout")
+	remotePeer := getopt.StringLong("remote-peer-address", 'r', "127.0.0.1", "Address or hostname of the remote peer (PFCP Agent)")
+	upfAddr := getopt.StringLong("upf-address", 'u', defaultUpfN3Address, "Address of the UPF (UP4)")
+	sessionCnt := getopt.IntLong("session-count", 'c', 1, "Set the amount of sessions to create, starting from 1 (included)")
+	ueAddrPool := getopt.StringLong("ue-address-pool", 'e', defaultUeAddressPool, "The IPv4 CIDR prefix from which UE addresses will be generated, incrementally")
+	NodeBAddr := getopt.StringLong("nodeb-address", 'g', defaultGNodeBAddress, "The IPv4 of (g/e)NodeBAddress")
+	verbosity := getopt.BoolLong("verbose", 'v', "Set verbosity level to debug")
+
+	optHelp := getopt.BoolLong("help", 0, "Help")
+
+	getopt.Parse()
+	if *optHelp {
+		getopt.Usage()
+		os.Exit(0)
+	}
+
+	// Flag checks and validations
+	if *verbosity {
+		level := log.DebugLevel
+		log.SetLevel(level)
+		log.Infof("Verbosity level set to: %v", level.String())
+	}
+
+	if *outputFile != "" {
+		fn := copyOutputToLogfile(*outputFile)
+		defer fn()
+	}
+
+	if *inputF != "" {
+		inputFile = *inputF
+	}
+
+	if *sessionCnt <= 0 {
+		log.Fatalf("Session count cannot be 0 or a negative number")
+	}
+	sessionCount = *sessionCnt
+
+	// IPs checks
+	nodeBAddress = net.ParseIP(*NodeBAddr)
+	if nodeBAddress == nil {
+		log.Fatalf("Could not retrieve IP address of (g/e)NodeB")
+	}
+
+	remotePeerAddress = net.ParseIP(*remotePeer)
+	if remotePeerAddress == nil {
+		address, err := net.LookupHost(*remotePeer)
+		if err != nil {
+			log.Fatalf("Could not retrieve hostname or address for remote peer: %s", *remotePeer)
+		}
+		remotePeerAddress = net.ParseIP(address[0])
+	}
+
+	upfAddress = net.ParseIP(*upfAddr)
+	if upfAddress == nil {
+		log.Fatalf("Error while parsing UPF address")
+	}
+
+	var err error = nil
+
+	_, _, err = net.ParseCIDR(*ueAddrPool)
```

Comment:
If we don't need the above line, you can simply use:

```suggestion
	_, _, err := net.ParseCIDR(*ueAddrPool)
```